Import: File is removed from cache before import

... now I'm glad that we added in that extra exception reporting. Likely same cause as #5880

https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/a4a8e719-c87e-4043-864a-884722cbc6dd

```
java.io.FileNotFoundException: /data/user/0/com.ichi2.anki/cache/AnKing_V6_pineapples.apkg: open failed: ENOENT (No such file or directory)
at libcore.io.IoBridge.open(IoBridge.java:455)
at java.io.RandomAccessFile.<init>(RandomAccessFile.java:247)
at org.apache.commons.compress.archivers.zip.ZipFile.<init>(ZipFile.java:18)
at org.apache.commons.compress.archivers.zip.ZipFile.<init>(ZipFile.java:4)
at org.apache.commons.compress.archivers.zip.ZipFile.<init>(ZipFile.java:1)
at com.ichi2.libanki.importer.AnkiPackageImporter.run(AnkiPackageImporter.java:4)
at com.ichi2.async.CollectionTask.doInBackgroundImportAdd(CollectionTask.java:7)
at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:38)
at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:1)
at android.os.AsyncTask$2.call(AsyncTask.java:304)
at java.util.concurrent.FutureTask.run(FutureTask.java:237)
at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
at java.lang.Thread.run(Thread.java:762)
Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
at libcore.io.Posix.open(Native Method)
at libcore.io.BlockGuardOs.open(BlockGuardOs.java:187)
at libcore.io.IoBridge.open(IoBridge.java:441)
... 14 more
```

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/importer/AnkiPackageImporter.java
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/importer/AnkiPackageImporter.java
@@ -60,7 +60,16 @@ public class AnkiPackageImporter extends Anki2Importer {
             String colname = "collection.anki21";
             try {
                 // extract the deck from the zip file
-                mZip = new ZipFile(new File(mFile));
+                try {
+                    mZip = new ZipFile(new File(mFile));
+                } catch (FileNotFoundException fileNotFound) {
+                    // The cache can be cleared between copying the file in and importing. This is temporary
+                    if (fileNotFound.getMessage().contains("ENOENT")) {
+                        mLog.add(getRes().getString(R.string.import_log_file_cache_cleared));
+                        return;
+                    }
+                    throw fileNotFound; //displays: failed to unzip
+                }
                 // v2 scheduler?
                 if (mZip.getEntry(colname) == null) {
                     colname = CollectionHelper.COLLECTION_FILENAME;

--- a/AnkiDroid/src/main/res/values/02-strings.xml
+++ b/AnkiDroid/src/main/res/values/02-strings.xml
@@ -161,7 +161,7 @@
     <string name="import_log_failed_copy_to">Failed to copy apkg to %s</string>
     <string name="import_log_failed_validate">apkg failed validation</string>
     <string name="import_log_insufficient_space">There is not enough space to unzip the package. Needed %1$d, available %2$d</string>
-    <string name="import_log_file_cache_cleared">Temporary error importing file.\nPlease try again</string>
+    <string name="import_log_file_cache_cleared">Error importing file, likely a cache clear during processing.\nPlease try again</string>
     <string name="import_succeeded_but_check_database">Data import succeeded but post-import cleanup failed. Run check database later. Root cause: %s</string>
     <string name="import_error_unhandled_request">Unable to process import request</string>
     <string name="import_error_exception">Failed to import package\n\n%s</string>

--- a/AnkiDroid/src/main/res/values/02-strings.xml
+++ b/AnkiDroid/src/main/res/values/02-strings.xml
@@ -161,6 +161,7 @@
     <string name="import_log_failed_copy_to">Failed to copy apkg to %s</string>
     <string name="import_log_failed_validate">apkg failed validation</string>
     <string name="import_log_insufficient_space">There is not enough space to unzip the package. Needed %1$d, available %2$d</string>
+    <string name="import_log_file_cache_cleared">Temporary error importing file.\nPlease try again</string>
     <string name="import_succeeded_but_check_database">Data import succeeded but post-import cleanup failed. Run check database later. Root cause: %s</string>
     <string name="import_error_unhandled_request">Unable to process import request</string>
     <string name="import_error_exception">Failed to import package\n\n%s</string>

